modules/web-console/frontend/app/configuration/generator/generator/JavaTransformer.service.spec.ts

```diff
@@ -35,67 +42,50 @@ suite('Java transformer tests', () => {
-                     **/
-                    public static void main(String[] args) throws Exception {
-                        try (Ignite ignite = Ignition.start("TestCfg.xml")) {
-                            System.out.println(">>> Loading caches...");
-                
-                            System.out.println(">>> Loading cache: testCache");
-                            ignite.cache("testCache").loadCache(null);
-                
-                            System.out.println(">>> All caches loaded!");
-                        }
-                    }
-                }
-                `
-        );
+        assert.equal(_generate(false), testData.CACHE_LOAD_WO_PERSISTENCE_CONTENT);
+        assert.equal(_generate(true), testData.CACHE_LOAD_WITH_PERSISTENCE_CONTENT);
+    });
 
-        assert.equal(_generate(true),
-            outdent`
-                public class TestClassName {
-                    /**
-                     * <p>
-                     * Utility to load caches from database.
-                     * <p>
-                     * How to use:
-                     * <ul>
-                     *     <li>Start cluster.</li>
-                     *     <li>Start this utility and wait while load complete.</li>
-                     * </ul>
-                     * 
-                     * @param args Command line arguments, none required.
-                     * @throws Exception If failed.
-                     **/
-                    public static void main(String[] args) throws Exception {
-                        try (Ignite ignite = Ignition.start("TestCfg.xml")) {
-                            ignite.cluster().active(true);
-                
-                            System.out.println(">>> Loading caches...");
-                
-                            System.out.println(">>> Loading cache: testCache");
-                            ignite.cache("testCache").loadCache(null);
-                
-                            System.out.println(">>> All caches loaded!");
-                        }
-                    }
-                }
-                `
+    test('Should generate valid list of imports', () => {
+        const configuration = cloneDeep(testData.TEST_CONFIGURATION);
+
+        _addCacheConfiguration(configuration, testData.TEST_CACHE);
+
+        const imports = filter(
+            [...IgniteJavaTransformer.collectConfigurationImports(configuration).values()],
+            (cls) => !startsWith(cls, 'java.lang.') && includes(cls, '.')
         );
+
+        assert.equal(testData.EXPECTED_IMPORTS.length, imports.length);
+
+        forEach(testData.EXPECTED_IMPORTS, (expectedImport) => assert.equal(true, imports.indexOf(expectedImport) >= 0));
     });
+
+    test('Should generate list of imports for big configuration without exceptions', () => {
+        const configuration = cloneDeep(testData.TEST_CONFIGURATION);
+
+        _addCacheConfiguration(configuration, testData.TEST_CACHE, 4000);
```

Comment:
How long does it take for this test to complete?

Reply:
From what @vsisko says, it takes ~8s on his computer. I think this is unacceptable, it doubles the unit test suite run time. We should do something about this.

Reply:
9 sec.

Reply:
@Klaster1  do you have any ideas what can be optimized?

Reply:
@akuznetsov-gridgain I see several options here:
1. Optimize the performance of unit test and application code that participate in the test. This might take a while, but there also might be some low hanging fruits. We should at least profile the run and see what takes so much time.
2. If option 1 does not yield good enough results or would take too long to execute, it should not be a part of unit test suite. I have no idea where it can be moved to. Maybe introduce optional unit tests that run only on CI? Please suggest your ideas.

Reply:
Optimized by @akuznetsov-gridgain